### PR TITLE
TELCODOCS-362: D/S Documentation & RN: SDN-2215/MPNETWORK-4:Track Kubernetes NMstate to be GA for the bare metal platform

### PR DESCRIPTION
**Pull request by johnwilkins**
Removes Technology Preview admonishment for K8S NMState on bare metal platforms.

Fixes: [TELCODOCS-362](https://issues.redhat.com/browse/TELCODOCS-362)

See https://issues.redhat.com/browse/TELCODOCS-362 for additional details.

Preview URL: https://deploy-preview-41108--osdocs.netlify.app/openshift-enterprise/latest/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html

https://deploy-preview-41108--osdocs.netlify.app/openshift-enterprise/latest/networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.html#virt-creating-interface-on-nodes_k8s_nmstate-updating-node-network-config

For release(s): 4.10
Signed-off-by: John Wilkins <jowilkin@redhat.com>

### CODE REVIEW

**Comment by racedo** [comment]
A few things come to mind:

1. To configure the DNS resolvers you need an interface, i.e. you can’t add a manifest that changes the DNS resolvers without specifying an interface along with it. Your example snippet has it already, I’d suggest being explicit in the description about this since this isn’t intuitive, i.e. if you were to do this in the OS you’d go to /etc/resolv.conf and just add your DNS servers there, no interface needed, but with knmstate you do need them and users don’t necessarily need to know beforehand.

2. You also needed to set auto-dns false if using DHCP on your interface or having a static IP and default gateway in your interface

3. You can’t modify br-ex or even use it as the interface you specify when changing your DNS resolvers. I don’t know why this happens (or whether it only happens with OVN) 

@cybertron would you know any details about 3?

@johnwilkins these points should be described in the docs for DNS

**Comment by cybertron** [comment]
>     3. You can’t modify br-ex or even use it as the interface you specify when changing your DNS resolvers. I don’t know why this happens (or whether it only happens with OVN)

This is only an OVN thing, and it's because there's a separate script that creates br-ex. If we modify it using kubernetes-nmstate then we will have two configuration mechanisms fighting each other. It's also non-trivial and error-prone to try to reproduce the exact br-ex configuration in nmstate since it's a declarative interface and you can't just modify an existing configuration without replicating the entire existing configuration in nmstate.

File: modules/virt-creating-interface-on-nodes.adoc [hunk 1]
```diff
@@ -41,12 +41,20 @@ spec:
               enabled: false
           port:
             - name: eth1
+    dns-resolver: <6>
```
Comment (cybertron) [inline modules/virt-creating-interface-on-nodes.adoc:44]: I don't think this will work. We need auto-dns: false in the bridge ipv4 section so there's a place to store the DNS settings.

File: modules/virt-example-nmstate-IP-management.adoc [hunk 2]
```diff
@@ -106,6 +110,14 @@ The following snippet sets DNS configuration on the host.
         - 8.8.8.8
 ...
 ----
+<1> You must configure an interface when configuring the DNS resolver. A manifest that changes the DNS resolver configuration and does not include an interface configuration will not work.
+
+<2> You must set the `auto-dns` configuration setting on the interface configuration to `false` if using DHCP on the interface, or if you configure a static IP address and a default gateway on the interface.
+
+[IMPORTANT]
+====
+You cannot use `br-ex`, an Open vSwitch brige, as the interface when configuring DNS resolvers.
```
Comment (cybertron) [inline modules/virt-example-nmstate-IP-management.adoc:119]: *bridge

I might also reword this as "You cannot use `br-ex`, an OVNKubernetes-managed Open vSwitch bridge" to make it clear that it's not all OVS bridges that are a problem.

File: modules/virt-example-nmstate-IP-management.adoc [hunk 3]
```diff
@@ -106,6 +110,14 @@ The following snippet sets DNS configuration on the host.
         - 8.8.8.8
 ...
 ----
+<1> You must configure an interface when configuring the DNS resolver. A manifest that changes the DNS resolver configuration and does not include an interface configuration will not work.
```
Comment (cybertron) [inline modules/virt-example-nmstate-IP-management.adoc:113]: I think I would combine these two points into something like:

> You must configure an interface with `auto-dns: false` or you must use static IP configuration on an interface in order for Kubernetes-NMState to store custom DNS settings.

Strictly speaking, if you had an interface with a static IP already you wouldn't need the interface section in the DNS policy. You just need _some_ interface that can store static DNS settings.

**Review by cybertron — CHANGES_REQUESTED** [review]

File: modules/ipi-install-network-requirements.adoc [hunk 4]
```diff
@@ -147,16 +147,16 @@ Define a consistent clock date and time format in each cluster node's BIOS setti
 You can reconfigure the control plane nodes to act as NTP servers on disconnected clusters, and reconfigure worker nodes to retrieve time from the control plane nodes.
 
 [discrete]
-== State-driven network configuration requirements (Technology Preview)
+== State-driven network configuration requirements
 
-{product-title} supports additional post-installation state-driven network configuration on the secondary network interfaces of cluster nodes using `kubernetes-nmstate`. For example, system administrators might configure a secondary network interface on cluster nodes after installation for a storage network.
+Installer-provisioned installation supports additional post-installation state-driven network configuration on the secondary network interfaces of cluster nodes using `kubernetes-nmstate`. For example, system administrators might configure a secondary network interface on cluster nodes after installation for a storage network.
```
Comment (djuran) [inline modules/ipi-install-network-requirements.adoc:152]: Why are we mixing IPI into this topic when all of this is happening post-installation.
As this is about post-installtion, I don't think we need to mention anythin in relation to the installers, do we?

Comment (rdoxenham) [reply modules/ipi-install-network-requirements.adoc:152]: Agreed, the fact that it's IPI or UPI shouldn't matter if this is all taken care of post-deployment. The same mechanisms for applying such configuration (either via kubernetes-nmstate or MachineConfig) should be available, regardless of the deployment mechanism.

File: modules/virt-creating-interface-on-nodes.adoc (hunk 1 above)
Comment (johnwilkins) [reply modules/virt-creating-interface-on-nodes.adoc:46]: Fixed.

**Comment by mjpytlak** [comment]
Peer review LGTM. @johnwilkins I do not have merge rights. I have the request out to other members of the peer review squad.

**Comment by jeana-redhat** [comment]
added peer review label for Mike and will merge+CP
